### DISCUSSION TITLE
1.6.0开源版本 force_release_deleted_file_fd_timeout 参数不生效

### DISCUSSION THREAD

**Discussion by lphu** [alibaba/ilogtail#1022] [Help]
logtail version: 1.6.0
force_release_deleted_file_fd_timeout: 120

尝试删除文件后, 文件句柄立即被释放。
![image](https://github.com/alibaba/ilogtail/assets/33591744/41d7e793-dbac-4f22-9797-92ac2987404a)

DEFINE_FLAG_INT32(force_release_deleted_file_fd_timeout,
                  "force release fd if file is deleted after specified seconds, no matter read to end or not",
                  120);
将源码中该值修改为120，重新编译打包后尝试仍是如此，请问现在开源版本是不支持该参数吗?

**Comment by messixukejia** [comment]
上面描述没太清楚预期是什么？可以再描述下。

代码里有就是支持的。参数意义参考下force_release_deleted_file_fd_timeout  https://help.aliyun.com/document_detail/32278.html?spm=a2c4g.28958.0.i1

**Comment by lphu** [comment]
想通过此参数配置来延长 容器被销毁时间。现在容器内的日志目录是通过emptyDir挂载，logtail使用daemonset方式监听emptyDir目录，想通过此参数配置来控制容器被销毁的时间来解决容器销毁时的日志丢失问题。

但是测试过程中发现此参数没有生效，即使配置后也会立即释放文件句柄信息。

**Comment by messixukejia** [comment]
句柄释放很多条件，影响因素很多。文件是否读完也是一个因素。你上面测试可以改成文件持续写入的场景下进行，不要是静态文件。

方便的话麻烦登记下使用场景吧https://github.com/alibaba/ilogtail/issues/693

**Reply by lphu** [reply]
我提供的case就是有一个shell脚本持续性写入的。

或者你们能提供一个logtail在文件删除时继续采集的例子吗?

**Reply by yyuuttaaoo** [reply]
这个参数限制的是最长强制释放时间。不能用来达到你所描述的目的。你目前的case ilogtail不能完整采集数据吗？
